### PR TITLE
V2

### PR DESCRIPTION
**Pull request by seymen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,39 +71,51 @@ Feature:
...
 1 scenario (1 passed)
 3 steps (3 passed)
 ```
 
 ## Step timeout
-Cucumber.js default step timeout is 5000ms. Follow [this guide](https://github.com/cucumber/cucumber-js#timeouts) to change it for your steps.
+Cucumber.js default step timeout is 5000ms. Add the following to features/support/init.js in order to change it:
+
+```javascript
+defineSupportCode(function({setDefaultTimeout}) {
```
Comment (seantdg) [inline README.md]: we could say something to indicate that this is in milliseconds, not seconds

File: README.md [hunk 2]
```diff
@@ -224,21 +232,22 @@ The following gherkin expressions are implemented in apickli source code [source
 ```
 GIVEN:
 	I set (.*) header to (.*)
-    I set cookie to (.*)
+    	I set cookie to (.*)
```
Comment (seantdg) [inline README.md]: damn you vim tabs vs spaces...

File: README.md [hunk 3]
```diff
@@ -252,10 +261,10 @@ THEN:
...
 	response body path (.*) should be (.*)
 	response body path (.*) should not be (.*)
-    	response body path (.*) should be of type array
-    	response body path (.*) should be of type array with length (\d+)
-    	response body should be valid according to schema file (.*)
-    	response body should be valid according to swagger definition (.*) in file (.*)
+   	response body path (.*) should be of type array
+   	response body path (.*) should be of type array with length (\d+)
+   	response body should be valid according to schema file (.*)
+   	response body should be valid according to openapi description (.*) in file (.*)
```
Comment (seantdg) [inline README.md]: definition vs description vs specification?

i prefer openapi specification

File: README.md [hunk 4]
```diff
@@ -264,15 +273,6 @@ THEN:
...
 ```
 
-The simplest way to adopt these expressions is to create a file named apickli-gherkin.js in features/step_definitions and extend the apickli/gherkin.js module.
-
-add the following to test/features/step_definitions/apickli-gherkin.js
-```javascript
-module.exports = require('apickli/apickli-gherkin');
-```
-
-If using Windows, follow this guide to create a symlink: [How-To Geek Guide](http://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/).
```
Comment (seantdg) [inline README.md]: i know we have removed this, but i had this issue again today with some developers. some windows advice around using symlinks and around cucumber-js vs cucumber.js (you must use a - in windows) might be useful

File: source/apickli/apickli-gherkin.js [hunk 5]
```diff
@@ -25,38 +26,38 @@ const callbackWithAssertion = function(callback, assertion) {
...
 };
 
-module.exports = function() {
-    this.registerHandler('BeforeScenario', function(scenario, callback) {
-        stepContext.scenario = scenario.getName;
-        callback();
-    });
+defineSupportCode(function({Before}) {
+    Before(function(scenarioResult, callback) {
+        // https://github.com/cucumber/cucumber-js/issues/891
```
Comment (seantdg) [inline source/apickli/apickli-gherkin.js]: do we want to include these comments?

File: source/apickli/apickli-gherkin.js [hunk 6]
```diff
@@ -25,38 +26,38 @@ const callbackWithAssertion = function(callback, assertion) {
...
+        // stepContext.scenario = scenario.getName;
 
-    this.registerHandler('BeforeStep', function(step, callback) {
-        stepContext.step = step.getName;
         callback();
     });
+});
 
-    this.Given(/^I set (.*) header to (.*)$/, function(headerName, headerValue, callback) {
+defineSupportCode(function({Given, When, Then}) {
```
Comment (seantdg) [inline source/apickli/apickli-gherkin.js]: Feel free to disagree with this statement...

In my mind, support code includes Before and After hooks, as well as environment code such as setting timeouts.
I don't think step definitions are usually defined as support code. (see [here](https://relishapp.com/cucumber/cucumber/docs/writing-support-code) ). 

Must they be inside defineSupportCode to work?

File: .travis.yml [hunk 7]
```diff
@@ -3,11 +3,8 @@ language: node_js
...
-  - npm install -g cucumber@^1.0.0
-  - npm install -g gulp
-  - cp source/apickli/apickli-gherkin.js source/test/features/step_definitions/
-script:
   - cd source
-  - gulp lint
-  - cucumber-js test/ --tags @core && gulp test
+  - npm install
+script:
+  - node_modules/.bin/gulp lint
```
Comment (seantdg) [inline .travis.yml]: does gulp lint only lint the code? or does it also run the tests?
we may want to include gulp test too

File: README.md (hunk 3 above)
Comment (seymen) [inline README.md]: I think description is the correct term. Specification is the spec for the description! :)

```
The OpenAPI Specification is a community driven, open specification within the Open API Initiative, a Linux Foundation Collaborative Project.

The OpenAPI Specification (OAS) defines a standard, programming language-agnostic interface description for REST APIs, which allows both humans and computers to discover and understand the capabilities of a service without requiring access to source code
```
https://github.com/OAI/OpenAPI-Specification

File: README.md (hunk 4 above)
Comment (seymen) [inline README.md]: Do we still need symlinking now that we have a way of actually requiring the apickli-gherkin module?

File: source/apickli/apickli-gherkin.js (hunk 5 above)
Comment (seymen) [inline source/apickli/apickli-gherkin.js]: let's keep it so people can understand why they are commented. Once the issue is fixed, we can remove it.

File: source/apickli/apickli-gherkin.js (hunk 6 above)
Comment (seymen) [inline source/apickli/apickli-gherkin.js]: If you are talking about the name `defineSupportCode`, then it is not our code/choice. That handler is defined by cucumber 3 api: https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/step_definitions.md

File: .travis.yml (hunk 7 above)
Comment (seymen) [inline .travis.yml]: only lints code.
gulp test doesn't work - see https://github.com/vgamula/gulp-cucumber/issues/42.
Once that issue is fixed, we can add `gulp test` back in. This is only temporary until that issue is fixed.

File: README.md (hunk 1 above)
Comment (seymen) [inline README.md]: done

File: README.md (hunk 4 above)
Comment (seantdg) [inline README.md]: agreed, i take it back

File: source/apickli/apickli-gherkin.js (hunk 6 above)
Comment (seantdg) [inline source/apickli/apickli-gherkin.js]: boo!

File: .travis.yml (hunk 7 above)
Comment (seantdg) [inline .travis.yml]: 👍

**Comment by seantdg** [comment]
ready for merge (however looks like there are upstream changes...)

**Pull request merged**